File: gcj2010QRA.hs
#!/usr/bin/env runghc
import Control.Monad
import Data.Bits

main = do
  c <- liftM lines getContents
  let ss = zipWith (\i r->"Case #"++show i++": "++r) [1..] $ solve c
  mapM_ putStrLn ss

solve (c:cs) = map u $ analyze cs
  where
    u [n,k] = if (k .&. g) == g then "ON" else "OFF"
      where g = (shift 1 n)-1
    analyze = map (map read . words) ::[String]->[[Int]]

File: gcj2010QRA.py
#!/usr/bin/env python
import sys

def solve():
    N,K = map(int,sys.stdin.readline().split())
    b = (1<<N)-1
    return "ON" if (K&b)==b else "OFF"

for x in range(int(sys.stdin.readline())):
    print("Case #"+str(x+1)+": "+str(solve()))
